### ISSUE TITLE
Decompiler does not work with member sorting and eclipse 4.10

### ISSUE THREAD

**Issue by JanecekPetr** [ecd-plugin/ecd#59]
After updating to eclipse 2018-12 (4.10, build id I20181206-0815), the decompiler crashes the editor when I try to open a class file. The reported exception:

```
java.lang.IllegalArgumentException
	at org.eclipse.jdt.core.util.CompilationUnitSorter.checkASTLevel(CompilationUnitSorter.java:62)
	at org.eclipse.jdt.core.util.CompilationUnitSorter.sort(CompilationUnitSorter.java:339)
	at org.eclipse.jdt.internal.corext.codemanipulation.SortMembersOperation.run(SortMembersOperation.java:330)
	at org.sf.feeling.decompiler.util.SortMemberUtil.sortMember(SortMemberUtil.java:90)
	at org.sf.feeling.decompiler.editor.BaseDecompilerSourceMapper.findSource(BaseDecompilerSourceMapper.java:222)
	at org.sf.feeling.decompiler.editor.DecompilerSourceMapper.findSource(DecompilerSourceMapper.java:72)
	at org.sf.feeling.decompiler.editor.JavaDecompilerClassFileEditor.doOpenBuffer(JavaDecompilerClassFileEditor.java:173)
	at org.sf.feeling.decompiler.editor.JavaDecompilerClassFileEditor.doOpenBuffer(JavaDecompilerClassFileEditor.java:148)
	at org.sf.feeling.decompiler.editor.JavaDecompilerClassFileEditor.doOpenBuffer(JavaDecompilerClassFileEditor.java:140)
	at org.sf.feeling.decompiler.editor.JavaDecompilerClassFileEditor.doSetInput(JavaDecompilerClassFileEditor.java:472)
	at org.eclipse.ui.texteditor.AbstractTextEditor.lambda$1(AbstractTextEditor.java:3135)
	at org.eclipse.jface.operation.ModalContext.runInCurrentThread(ModalContext.java:440)
	at org.eclipse.jface.operation.ModalContext.run(ModalContext.java:356)
```

My JDT version: 3.16.0.v20181206-1038
My ECD version: 3.1.1

**Comment by JanecekPetr** [comment]
Hm! This actually looks like a JDT bug.

The [`SortMembersOperation.run()`](https://github.com/eclipse/eclipse.jdt.ui/blob/master/org.eclipse.jdt.ui/core%20extension/org/eclipse/jdt/internal/corext/codemanipulation/SortMembersOperation.java#L330) operation calls [`CompilationUnitSorter.sort()`](https://github.com/eclipse/eclipse.jdt.core/blob/master/org.eclipse.jdt.core/model/org/eclipse/jdt/core/util/CompilationUnitSorter.java#L331) with [`AST.JLS11`](https://git.eclipse.org/c/jdt/eclipse.jdt.ui.git/plain/org.eclipse.jdt.core.manipulation/core%20extension/org/eclipse/jdt/internal/corext/dom/IASTSharedValues.java), but [it only supports JLS10](https://github.com/eclipse/eclipse.jdt.core/blob/master/org.eclipse.jdt.core/model/org/eclipse/jdt/core/util/CompilationUnitSorter.java#L52).

I'll try to file a bug against JDT and will close this one once the new bug is accepted.

**Comment by RobertZenz** [comment]
Thanks for the all the digging.

I'm torn whether we should provide a hotfix on that one. On the one end it would be a good idea to make sure that ECD works everywhere, on the other side I ain't got the time currently to look into this and this will go away in three months anyway...So I'm in favor of the default-ticket-tactic, letting it slide until it resolves itself.

**Comment by JanecekPetr** [comment]
This has been fixed in eclipse 2020-03 (4.15), for more information see https://bugs.eclipse.org/bugs/show_bug.cgi?id=543073.
